lib/ansible/plugins/strategy/free.py

```diff
@@ -76,6 +80,7 @@ def run(self, iterator, play_context):
         list again, which would end up favoring hosts near the beginning of the
         list.
         '''
+        self.iterator = iterator
```

Comment:
I kind of question this line here.  I understand what you are doing and trying to achieve, but it feels kind of like a hack to inject this into class at this point...working around a shortcoming of the class.  This makes me feel like that should be something that is addressed.  Not sure, just offering opinions on my immediate reaction.

Reply:
@sivel Thanks for commenting. I'm not happy with this line either and it's why I attempted to use the TQM at first. I could instead pass the iterator to the filtering function. Do you think that would be better?

Reply:
My only concern with that is that it could break 3rd party strategy plugins.